tutorials/high_performance_networking/README.md

```diff
@@ -0,0 +1,2171 @@
+
+        ```
+        hugepages-1048576kB
+        hugepages-2048kB
+        hugepages-32768kB
+        hugepages-64kB
+        ```
+
+        ```
+        HugePages_Total:       0
+        HugePages_Free:        0
+        HugePages_Rsvd:        0
+        HugePages_Surp:        0
+        Hugepagesize:       2048 kB
+        Hugetlb:               0 kB
+        ```
+
+    And your huge page mount points:
+
+    ```bash
+    mount | grep huge
+    ```
+
+    ??? abstract "See an example output"
+
+        The default huge pages are mounted on `/dev/hugepages` with a page size of 2M.
+
+        ```
+        hugetlbfs on /dev/hugepages type hugetlbfs (rw,relatime,pagesize=2M)
+        ```
+
+**As a rule of thumb, we recommend to start with 3 to 4 GB of total huge pages, with an individual page size of 500 MB to 1 GB** (per system availability).
+
+There are two ways to allocate huge pages:
+
+- in the kernel bootline (recommended to ensure contiguous memory allocation) or
+- dynamically at runtime (risk of fragmentation for large page sizes)
+
+The example below allocates 3 huge pages of 1GB each.
+
+=== "Kernel bootline"
+
+    Add the flags below to the `GRUB_CMDLINE_LINUX` variable in `/etc/default/grub`:
+
+    ```bash
+    default_hugepagesz=1G hugepagesz=1G hugepages=3
+    ```
+
+    ??? info "Show explanation"
+
+        - `default_hugepagesz`: the default huge page size to use, making them available from the default mount point, `/dev/hugepages`.
+        - `hugepagesz`: the size of the huge pages to allocate.
+        - `hugepages`: the number of huge pages to allocate.
+
+    Then rebuild your GRUB configuration and reboot:
+
+    ```bash
+    sudo update-grub
+    sudo reboot
+    ```
+
+=== "Runtime"
+
+    Allocate the 3x 1GB huge pages:
+
+    === "hugeadm"
+
+        ```bash
+        sudo hugeadm --pool-pages-min 1073741824:3
+        ```
+
+    === "vanilla"
+
+        ```bash
+        echo 3 | sudo tee /sys/kernel/mm/hugepages/hugepages-1048576kB/nr_hugepages
+        ```
+
+    Create a mount point to access the 1GB huge pages pool since that is not the default size on that system. We will name it `/mnt/huge` here.
+
+    === "One-time"
+
+        ```bash
+        sudo mkdir -p /mnt/huge
+        sudo mount -t hugetlbfs -o pagesize=1G none /mnt/huge
+        ```
+
+    === "Persistent"
+
+        ```bash
+        echo "nodev /mnt/huge hugetlbfs pagesize=1G 0 0" | sudo tee -a /etc/fstab
+        sudo mount /mnt/huge
+        ```
+
+    !!! note
+
+        If you work with containers, remember to mount this directory in your container as well with `-v /mnt/huge:/mnt/huge`.
+
+
+Rerunning the initial commands should now list 3 hugepages of 1GB each. 1GB will be the default huge page size if updated in the kernel bootline only.
+
+### 3.5 Isolate CPU cores
+
+!!! note
+
+    This optimization is less impactful when using the `gpunetio` backend since the GPU polls the NIC.
```

Comment:
Do we need to define what isolation means here?

Reply:
Sounds good. Can you suggest some text?

Reply:
"Isolating a CPU in Linux prevents unwanted user or kernel threads from running on the selected CPU. This decreases context switching and avoids noisy neighbor problems that can increase latency in your application."

Reply:
chose this:

```md
The CPU interacting with the NIC to route packets means that any perturbation can have a large impact on latency, especially with smaller packet/batch sizes requiring more frequent work. Isolating a CPU in Linux prevents unwanted user or kernel threads from running on the selected CPU. This decreases context switching and avoids noisy neighbor problems that can increase latency in your application.

We recommend isolating the CPU cores you will select to interact with the NIC (defined in the `advanced_network` configuration [described later](#51-understand-the-configuration-parameters) in this tutorial). This is done by setting additional flags on the kernel bootline.

Reply:
I was thinking 

```
The CPU interacting with the NIC to route packets means that any perturbation can have a large impact on latency, especially with smaller packet/batch sizes requiring more frequent work.

To avoid dropping packets, we recommend isolating the CPU cores that are interacting with the NIC (defined in the advanced_network configuration [described later](https://github.com/nvidia-holoscan/holohub/pull/718#51-understand-the-configuration-parameters) in this tutorial). Isolating a CPU in Linux prevents unwanted user or kernel threads from running on the selected CPU. This decreases context switching and avoids noisy neighbor problems that can increase latency in your application. This is done by setting additional flags on the kernel bootline.
```